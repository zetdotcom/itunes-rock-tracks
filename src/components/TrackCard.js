import React from 'react';

export default function TrackCard({ trackName, artist, price, artworkUrl, id }) {
	return (
		<div data-testid='track-card' style={{ border: '1px solid red', height: '100%' }}>
			<div>
				<img width='100%' height='100%' src={artworkUrl} alt={trackName} />
			</div>
			<div>{trackName}</div>
			<div>{artist}</div>
			<div>{price}</div>
		</div>
	);
}
