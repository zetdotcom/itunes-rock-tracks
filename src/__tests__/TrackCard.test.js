import React from 'react';
import { createMemoryHistory } from 'history';
import { render, cleanup, fireEvent, waitForElement } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TrackCard from '../components/TrackCard';
import { Router } from 'react-router-dom';

const props = propsOverride => ({
	trackName: 'track name',
	artist: 'artist',
	price: '1.28',
	artworkUrl: '',
	id: 1,
	...propsOverride,
});

const history = createMemoryHistory();
const renderComponent = () =>
	render(
		<Router history={history}>
			<TrackCard {...props()} />
		</Router>
	);

afterEach(cleanup);

describe('TEST TrackCard', () => {
	test('should match snapshot', () => {
		const { asFragment } = renderComponent();
		expect(asFragment).toMatchSnapshot();
	});

	test('should format time, price and date correctly', async () => {
		const { findByText } = renderComponent();
		const price = await waitForElement(() => findByText('$1.28'));

		expect(price).toBeInTheDocument();
	});
});
