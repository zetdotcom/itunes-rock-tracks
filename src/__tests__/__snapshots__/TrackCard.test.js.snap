// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST TrackCard should match snapshot 1`] = `[Function]`;
